code/mob/living/critter/small_animal.dm

```diff
@@ -447,22 +459,113 @@ ABSTRACT_TYPE(/mob/living/critter/small_animal)
+		if (src.ai.enabled && ispug(user) && prob(10))
+			src.recent_pug_pet = TRUE
+			SPAWN(15 SECONDS)
+				src?.recent_pug_pet = FALSE
+			src.ai.priority_tasks += src.ai.get_instance(/datum/aiTask/sequence/goalbased/critter/attack, list(src, src.ai.default_task))
+			src.ai.interrupt()
+			src.visible_message("<span class='notice'>[src] recoils and hisses at [user]'s attempt to pet them, then goes for the jugular!</span>")
+			playsound(src.loc, 'sound/voice/animal/cat_hiss.ogg', 50, 1)
+
+	Crossed(atom/movable/M as mob)
+		..()
+		if (!isalive(src))
+			return
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			if(prob(10) && src.is_annoying)
+				src.visible_message("<span class='combat'>[src] weaves around [H]'s legs and trips [him_or_her(H)]!</span>")
+				H.setStatus("resting", duration = INFINITE_STATUS)
+				H.force_laydown_standup()
+			else if (prob(4) && ishuman(M))
+				boutput(src, "<span class='notice'>You weave around [H] to [pick("show your affection!", "get them to feed you.", "annoy them for no reason in particular.")]</span>")
+				boutput(H, "<span class='notice'>[src] weaves around you, waving their tail around. A bunch of hair clings to your clothes and some gets in your nose.</span>")
+				H.emote("sneeze")
+		return
+
+	seek_target(range)
+		. = list()
+		if (src.recent_pug_pet)
+			for (var/mob/living/carbon/human/H in hearers(range, src))
+				if (ispug(H) && isalive(H))
+					. += H
+					return
+		for (var/mob/living/critter/small_animal/mouse/C in hearers(range, src))
+			if (isdead(C)) continue
+			. += C
+		for (var/obj/critter/livingtail/C in hearers(range, src))
+			. += C
+
+		if (length(.) && prob(20))
+			playsound(src.loc, 'sound/voice/animal/cat_hiss.ogg', 50, 1)
+			src.visible_message("<span class='alert'>[src] hisses!</span>")
+
+	critter_attack(var/the_target)
+		if (istype(the_target, /obj/critter))
+			var/obj/critter/C = the_target
+			if (C.health <= 0 && C.alive)
+				playsound(src.loc, 'sound/impact_sounds/Generic_Hit_1.ogg', 50, 1, -1)
+				C.health -= 2
+		else if (istype(the_target, /mob))
+			var/mob/target = the_target
+			if(istype(target, /mob/living/critter/small_animal/mouse/weak/mentor) && prob(90))
+				src.visible_message("<span class='combat'><B>[src]</B> tries to bite [target] but \the [target] dodges [pick("nimbly", "effortlessly", "gracefully")]!</span>")
+				return
+			if ((src.catnip || prob(2) ) && (last_claw_fury < (TIME + 15 SECONDS)))
+				var/attackCount = rand(5, 9)
+				var/iteration = 0
+				while (iteration <= attackCount && (get_dist(src, target) <= 1))
+					if (iteration == 0)
+						target.setStatus("weakened", 2 SECONDS)
+						src.visible_message("<span class='combat'>[src] [pick("starts to claw the living <b>shit</b> out of ", "unleashes a flury of claw at ")] [target]!</span>")
+					random_brute_damage(target, src.attack_damage + 2, 1)
+					playsound(src.loc, 'sound/impact_sounds/Flesh_Tear_3.ogg', 60, 1)
+					iteration ++
+					sleep(0.3 SECONDS)
```

Comment:
this sleep should probably be behind a spawn so it doesn't lock up the mob loop or something

Reply:
Donezo, do tell me if i did it backwards